Date fix

benblack769 opened the pull request:
Minor server fixes. 

resolves #34 

This should allow us to deploy before merging in admin_redo.

benblack769 committed:
date fix to regular upload

benblack769 committed:
date revert and image edit fix

rekennelly commented on strabo/private_helper.py:
```diff
@@ -24,14 +24,18 @@ def fill_interest_point(ip,image_ids,form_title,form_body,form_geo_obj,form_laye
     '''
-    if a flask.files object is passed in, then  
+    If a flask.files object is passed in, then it saves the new file and stores the new information in the database.
+    If the image row already stored infromation about an old image (the image was edited), then that old image is deleted.
```
should say "information" not infromation

rekennelly commented on strabo/private_helper.py:
```diff
@@ -24,14 +24,18 @@ def fill_interest_point(ip,image_ids,form_title,form_body,form_geo_obj,form_laye
+def fill_image(image,form_file_obj,form_descrip,year,month):
     '''
-    if a flask.files object is passed in, then  
+    If a flask.files object is passed in, then it saves the new file and stores the new information in the database.
```
What is the flask.files object?

benblack769 replied:
I added a link to clear this up in the docstring. http://flask.pocoo.org/docs/0.11/patterns/fileuploads/.

Basically it is what the client sends to the server when you upload a file, and it allows you to save that file somewhere.

rekennelly commented on strabo/templates/private/upload_images.html:
```diff
@@ -24,19 +24,15 @@
             <div class="form-group">
-            <label for="year">Date Taken:</label><br>
+            <label for="year">Date Taken [mm]-[yyyy]:</label><br>
             <i>Images are ordered by date when displayed.</i>
```
What does this mean? I know this wasn't a change in this PR, but I don't think this would be very clear to a user.

benblack769 replied:
Yeah, that is programming language creeping into the UI there. Whoops.

My goal here is to tell the administrator that images are displayed in a narrative fashion, where images with earlier dates are displayed before those with later dates. This is so that they know that dates do matter, even if they don't directly display anywhere.

How about just that: "Images are displayed in a narrative fashion; those with earlier dates are displayed before those with later dates."

kamba16 commented on strabo/private_helper.py:
```diff
@@ -24,14 +24,18 @@ def fill_interest_point(ip,image_ids,form_title,form_body,form_geo_obj,form_laye
+    image.taken_at = datetime.date(utils.safe_pos_int_conv(year),utils.safe_pos_int_conv(month),1)
     image.description = form_descrip
 
     if form_file_obj:
```
Why `form_file_obj`? Why not just `form_file`? It seems like that would be more clear, and consistent with `form_descrip`. Also, this may not be a big deal but you sometimes use "desc" when naming variables/methods to refer to description and other times you use "descrip." This doesn't matter for the functionality of the code but it is an inconsistency that puzzles me... I wonder if it would be better if we standardized those going forward.

benblack769 replied:
form_file is much better. I guess I wanted to differentiate it from the other form arguments, which are all strings, but it is probably clear enough on its own. It is also unbelievably bad Hungarian notation.

For name abbreviations, there is an entire literature on how and when and how much to abbreviate, and I am definitely not an expert, but basically you can think of it as 3 different versions of the same word
- description (11 characters)
- descrip (7 characters)
- desc (4 characters)

I don't think I do this in the code, but potentially, I might just type "d" (1 character)

Fewer characters means less clear and more ambiguous, but shorter, which is easier to type and sometimes easier to read. 

I won't claim that this is a terribly good example, and I often am inconsistent for no good reason, but I do think it important in general to have multiple abbreviations for certain words or phrases. In the strabo code, I tried to lean towards really long names for maximum clarity, and if it is confusing, then we can change desc to descrip, but I think you may often find it rather verbose as you get more experience.

kamba16 replied:
I think that desc is fine, but you use descrip above.

benblack769 replied:
Sorry, I thought that descrip was far more clear than desc. To me, desc can easily be thought to be all sorts of words like "descendant". If you think that desc is more clear than descrip then the argument above doesn't work, and desc should always be used unless you want to type out the whole 11 characters of "description".

kamba16 commented on strabo/private_helper.py:
```diff
@@ -24,14 +24,18 @@ def fill_interest_point(ip,image_ids,form_title,form_body,form_geo_obj,form_laye
+
         image.filename = file_writing.make_filename(form_file_obj.filename)
         file_writing.save_image_files(form_file_obj,image.filename)
         image.width,image.height = image_processing.get_dimentions(image.filename)
```
`dimentions` should be `dimensions`.

kamba16 commented:
"Images are displayed chronologically by year" or "Images are displayed chronologically by the year in which they were taken."

kamba16 commented:
So you wrote your own form validation (`upload_ips.html` and `upload_ips.js`)? I was earlier looking at Bootstrap's form validation code [here](http://www.w3schools.com/bootstrap/bootstrap_forms_inputs2.asp) and I'm wondering if you used or considered any of that code.

benblack769 commented:
I like "Images are displayed chronologically by year".

I think I tried the bootstrap form thingy and I think it didn't work. If it did ever work, then it would definitely make things look nicer, but it would not replace anything I have already done. (stop the form from submitting on bad input and display helpful messages to describe what went wrong).

benblack769 committed:
fixed naming, bugs, and user description

benblack769 committed:
fixed names and added link for clarification

benblack769 commented:
- Changed to "Images are displayed chronologically by year".
- Changed form_file_obj to obj
- Changed infromation to information.
- Added docstring link to clear up form.files.

Hope that clears up your concerns. 

Ava, do you still think descrip should be changed to desc? If so I can change that too.

benblack769 committed:
extract_extension removed

benblack769 commented:
Also deleted extract_name_extension according to the slack post and replaced it with os.path.splitext() in accordance with the slack post.

kamba16 commented on add_database.py:
```diff
@@ -52,24 +52,25 @@ def make_interest_point(image_ids,form_title,form_body,form_geo_obj,form_layer,f
+def make_image(form_file_obj,form_descrip,year,month):
     img = schema.Images()
-    private_helper.fill_image(img,form_file_obj,form_descrip,year,month,day)
+    private_helper.fill_image(img,form_file_obj,form_descrip,year,month)
```
should these be `form_file` as well, as opposed to `form_file_obj`?

kamba16 commented:
Thanks for making these changes!! I actually don't mind desc or descrip
either way - my comment was mainly due to the inconsistency (here we have
descrip, but in popup.js [I think] we have img_desc), but if mixing them is
standard practice then it is totally fine with me.

On Tue, Aug 9, 2016 at 11:29 PM, weepingwillowben notifications@github.com
wrote:

> Also deleted extract_name_extension according to the slack post and
> replaced it with os.path.splitext() in accordance with the slack post.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> https://github.com/reed-college/strabo/pull/57#issuecomment-238777134,
> or mute the thread
> https://github.com/notifications/unsubscribe-auth/AQliqF7xxmiTyWdJ2qLDKZM5WheMfMRIks5qeW_agaJpZM4JgTzj
> .

benblack769 committed:
fixed form_file_obj name

benblack769 commented:
Well, it is standard practice for me. I don't really examine other people's naming schemes, so I am not sure, but I feel like it works. And descr should be gone.
